### ISSUE TITLE
-gc boehm does not work with MSVC

### ISSUE THREAD

**Issue by Wertzui123** [vlang/v#12451]
**V version:** V 0.2.4 61b99e1
**OS:** Windows

**What did you do?**
```batch
v -cc msvc -gc boehm run main.v
```
main.v:
```v
fn main(){
	println("Hello World!")
}
```

**What did you expect to see?**
Successful compilation and then `Hello World!`.
 
**What did you see instead?**
A lot of these "undeclared identifier" errors in different lines of gc.c, e.g.:
```
path\to\v/thirdparty/libgc/gc.c(11718): error C2065: "__ATOMIC_RELAXED": nichtdeklarierter Bezeichner
```

**Comment by Ekopalypse** [comment]
For a workaround see #13133

**Comment by emily33901** [comment]
> Fixed in 0.3

I'm still getting this

```

PS C:\src\vproto> v version
V 0.3.0 d8b0df1

```

```v
module main

fn main() {
	println("Wow")
}

```

```
PS C:\src\vproto> v -cc msvc -gc boehm run test.v
C:\src\v\thirdparty\libgc\gc.obj not found, building it (with msvc)...
msvc: failed to build a thirdparty object; cmd: "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\bin\Hostx64\x64\cl.exe" /NOLOGO /volatile:ms /MDd /D_DEBUG /DGC_THREADS=1 /DGC_BUILTIN_ATOMIC=1 /DGC_NOT_DLL=1 /DGC_WIN32_THREADS=1 -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22000.0\ucrt" -I "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.32.31326\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22000.0\um" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22000.0\shared" -I "C:\src\v\thirdparty\libgc\include" /c "C:\src\v/thirdparty/libgc/gc.c" /Fo"C:\src\v\thirdparty\libgc\gc.obj"
builder error: Microsoft (R) C/C++ Optimizing Compiler Version 19.32.31332 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

cl : Command line warning D9002 : ignoring unknown option '/NOLOGO'
gc.c
C:\src\v/thirdparty/libgc/gc.c(12204): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(12243): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(12303): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(12329): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(12350): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(13160): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(13161): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(13571): error C2065: '__ATOMIC_SEQ_CST': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(13616): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(15938): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(15993): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(16025): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(16059): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(16750): error C2065: '__ATOMIC_ACQUIRE': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(16754): error C2065: '__ATOMIC_RELEASE': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(21850): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(22472): error C2065: '__ATOMIC_RELAXED': undeclared identifier
C:\src\v/thirdparty/libgc/gc.c(22601): error C2065: '__ATOMIC_RELAXED': undeclared identifier
```
